tests/foreman/virtwho/test_api_virtwho.py

```diff
@@ -97,18 +114,19 @@ def test_positive_deploy_configure_by_id(self):
             (guest_name, 'product_id={} and type=STACK_DERIVED'.format(
                 self.vdc_physical))]
         for hostname, sku in hosts:
-            host = entities.Host().search(
-                query={'search': "{0}".format(hostname)})[0]
-            subscriptions = entities.Subscription().search(
-                query={'search': '{0}'.format(sku)})
-            vdc_id = subscriptions[0].id
+            if 'type=NORMAL' in sku:
+                subscriptions = entities.Subscription().search(
+                    query={'search': '{0}'.format(sku)})
+                vdc_id = subscriptions[0].id
             if 'type=STACK_DERIVED' in sku:
-                for item in subscriptions:
-                    item = item.read_json()
-                    if hypervisor_name.lower() in item['hypervisor']['name']:
-                        vdc_id = item['id']
-                        break
-            entities.HostSubscription(host=host.id).add_subscriptions(
+                vdc_id = self._get_guest_bonus(hypervisor_name, sku)
+            host, time = wait_for(entities.Host().search,
```

Comment:
Why do you need `wait_for` to get the host? the hosts(hypervisor and guest) should be already existing after running ` deploy_configure_by_command`

Reply:
Not really, I found a lot of times about getting host failed(especially when I change the type of hypervisors), the error info was "index out of range", and the log showed that search result is `[]`, so there is a probability that we get the hosts failed.

Reply:
Failed to search hypervisor or guest? if hypervisor, which one? hyperv?

Reply:
From my debug, the error often came up when I change the Esx to Libvirt(or another hypervisor which has not registered to the Satellite yet, but not always), and when I rerun the case, it succeeded.